File: Basic/Second Largest.py
def secLargest(arr):
    first = sec = float('-inf')
    for i in arr:
       if i > first:
           sec = first
           first = i
       elif i > sec and i != first:
            sec = i
    return sec          
arr = list(map(int,input().split()))
print(secLargest(arr))

File: Basic/Second Largest.java
import java.util.*;
class Main {
    public static int secLargest(int[] arr){
        int first = Integer.MIN_VALUE;
        int sec = Integer.MIN_VALUE;
        for(int i = 0 ; i < arr.length;i++){
           if(arr[i]>first){
               sec=first;
               first = arr[i];
           }else if(arr[i]>sec && arr[i]!=first){
               sec=arr[i];
           }
        }
        return sec;
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr= new int[n];
       for(int i = 0 ; i < n ; i++){
           arr[i]=sc.nextInt();
       }
        System.out.println(secLargest(arr));
    }
}
